vernac/declare.ml

```diff
@@ -1990,18 +1990,20 @@ let get_keep_admitted_vars =
     ~key:["Keep"; "Admitted"; "Variables"]
     ~value:true
 
-let compute_proof_using_for_admitted proof typ pproofs =
-  if not (get_keep_admitted_vars ()) then None
-  else match get_used_variables proof, pproofs with
-    | Some _ as x, _ -> x
-    | None, pproof :: _ ->
-      let env = Global.env () in
-      let ids_typ = Environ.global_vars_set env typ in
-      (* [pproof] is evar-normalized by [partial_proof]. We don't
-         count variables appearing only in the type of evars. *)
-      let ids_def = Environ.global_vars_set env (EConstr.Unsafe.to_constr pproof) in
-      Some (Environ.really_needed env (Id.Set.union ids_typ ids_def))
-    | _ -> None
+let compute_proof_using_for_admitted proof typ iproof =
+  if not (get_keep_admitted_vars ()) || not (Global.sections_are_opened()) then None
+  else match get_used_variables proof with
+    | Some _ as x -> x
+    | None ->
+      match Proof.partial_proof iproof with
+      | pproof :: _ ->
+        let env = Global.env () in
+        let ids_typ = Environ.global_vars_set env typ in
+        (* [pproof] is evar-normalized by [partial_proof]. We don't
+           count variables appearing only in the type of evars. *)
+        let ids_def = Environ.global_vars_set env (EConstr.Unsafe.to_constr pproof) in
+        Some (Environ.really_needed env (Id.Set.union ids_typ ids_def))
+      | [] -> None
```

Comment:
```suggestion
let compute_proof_using_for_admitted proof typ iproof =
  if get_keep_admitted_vars () && Global.sections_are_opened () then
    match get_used_variables proof with
    | Some _ as x -> x
    | None ->
      match Proof.partial_proof iproof with
      | pproof :: _ ->
        let env = Global.env () in
        let ids_typ = Environ.global_vars_set env typ in
        (* [pproof] is evar-normalized by [partial_proof]. We don't
           count variables appearing only in the type of evars. *)
        let ids_def = Environ.global_vars_set env (EConstr.Unsafe.to_constr pproof) in
        Some (Environ.really_needed env (Id.Set.union ids_typ ids_def))
      | [] -> None
  else None
```

Reply:
The negation and if made it a bit confusing to read. Hopefully this is clearer.

Reply:
I prefer having the short case first